src/frontend/src/optimizer/rule/unify_first_last_value_rule.rs

```diff
@@ -0,0 +1,306 @@
+///
+/// This rule is particularly beneficial for streaming queries as it reduces the number of states maintained.
+/// It specifically targets `FIRST_VALUE` and `LAST_VALUE` aggregation functions that have identical ordering clauses.
+///
+/// # Example transformation:
+///
+/// ## Before:
+/// ```sql
+/// SELECT
+///   LAST_VALUE(col1 ORDER BY col2) AS last_col1,
+///   LAST_VALUE(col3 ORDER BY col2) AS last_col3,
+///   LAST_VALUE(col4 ORDER BY col2) AS last_col4,
+///   LAST_VALUE(col5 ORDER BY col2) AS last_col5
+/// FROM table_name
+/// GROUP BY group_col;
+/// ```
+///
+/// ## After:
+/// ```sql
+/// SELECT
+///   (unified_last).f0 AS last_col1,
+///   (unified_last).f1 AS last_col3,
+///   (unified_last).f2 AS last_col4,
+///   (unified_last).f3 AS last_col5
+/// FROM (
+///   SELECT
+///     LAST_VALUE(ROW(col1, col3, col4, col5) ORDER BY col2) AS unified_last
+///   FROM table_name GROUP BY group_col
+/// ) sub;
+/// ```
+///
+/// # Plan transformation:
+///
+/// ## Before:
+/// ```text
+/// LogicalAgg [group_col]
+///  ├─agg_calls:
+///  │  ├─ LAST_VALUE(col1 ORDER BY col2)     -- State 1
+///  │  ├─ LAST_VALUE(col3 ORDER BY col2)     -- State 2
+///  │  ├─ LAST_VALUE(col4 ORDER BY col2)     -- State 3
+///  │  └─ LAST_VALUE(col5 ORDER BY col2)     -- State 4
+///  └─LogicalScan { table: table_name }
+/// ```
+///
+/// ## After:
+/// ```text
+/// LogicalProject
+///  ├─exprs: [(unified_last).f0, (unified_last).f1, (unified_last).f2, (unified_last).f3]
+///  └─LogicalAgg [group_col]
+///     ├─agg_calls:
+///     │  └─ LAST_VALUE(ROW(col1,col3,col4,col5) ORDER BY col2)  -- Single State!
+///     └─LogicalProject
+///        ├─exprs: [group_col, col1, col2, col3, col4, col5, ROW(col1,col3,col4,col5)]
+///        └─LogicalScan { table: table_name }
+/// ```
+///
+/// The key benefit: **4 aggregation states → 1 aggregation state** for streaming performance!
+pub struct UnifyFirstLastValueRule {}
+
+impl Rule<Logical> for UnifyFirstLastValueRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let agg = plan.as_logical_agg()?;
+
+        let calls = agg.agg_calls();
+        if calls.len() < 2 {
+            // Need at least 2 calls to merge
+            return None;
+        }
+
+        // Group calls by their "pattern" (agg_type + order_by + distinct + filter)
+        let mut pattern_groups: HashMap<AggPattern, Vec<(usize, &PlanAggCall)>> = HashMap::new();
+
+        for (idx, call) in calls.iter().enumerate() {
+            if !self.is_supported_agg_type(&call.agg_type) {
+                continue;
+            }
+
+            // Only support single input aggregations for now
+            if call.inputs.len() != 1 {
+                continue;
+            }
+
+            let pattern = AggPattern {
+                agg_type: call.agg_type.clone(),
+                order_by: call.order_by.clone(),
+                distinct: call.distinct,
+                filter: call.filter.clone(),
+            };
```

Comment:
Shall we destruct `AggCall` here, so that we will be notified to update here if there's any change on fields of `AggCall`?